torch/distributed/algorithms/fsdp_comm_hooks/allreduce_hook.py

```diff
@@ -0,0 +1,49 @@
+import torch
+import torch.distributed as dist
```

Comment:
could we change the fsdp_comm_hooks to a general name like '_comm_hooks'? so that the DDP common hook can be merged here?

Reply:
Thanks for the suggestion, moved to a new folder.